python/ray/data/tests/test_execution_optimizer.py

```diff
@@ -967,6 +967,10 @@ def test_from_huggingface_operator(
 
 
 def test_from_huggingface_e2e(ray_start_regular_shared, enable_optimizer):
+    # Since `FromHuggingFace` reads from in-memory data, it will
+    # not record usage of the operator through the streaming executor.
+    # Therefore, in this test, we don't use the `_check_usage_record`
+    # method to check operator usage.
```

Comment:
I feel this is not right, can we double check? other method such as `from_items` also reads from in-memory data.